homeassistant/components/point/__init__.py

```diff
@@ -70,62 +82,102 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+            entry,
+            data={
+                **entry.data,
+                "auth_implementation": DOMAIN,
+                CONF_TOKEN: {
+                    **entry.data[CONF_TOKEN],
+                    "expires_at": 0,
+                },
+                "imported": True,
+            },
         )
 
-    session = PointSession(
-        async_get_clientsession(hass),
-        entry.data["refresh_args"][CONF_CLIENT_ID],
-        entry.data["refresh_args"][CONF_CLIENT_SECRET],
-        token=entry.data[CONF_TOKEN],
-        token_saver=token_saver,
+    implementation = (
+        await config_entry_oauth2_flow.async_get_config_entry_implementation(
+            hass, entry
+        )
     )
+    session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
+    auth = api.AsyncConfigEntryAuth(
+        aiohttp_client.async_get_clientsession(hass), session
+    )
+    entry.runtime_data = auth
+
     try:
-        # the call to user() implicitly calls ensure_active_token() in authlib
-        await session.user()
-    except ConnectTimeout as err:
-        _LOGGER.debug("Connection Timeout")
+        await auth.async_get_access_token()
+    except ClientResponseError as err:
+        if err.status in {HTTPStatus.UNAUTHORIZED, HTTPStatus.FORBIDDEN}:
+            raise ConfigEntryAuthFailed from err
+        raise ConfigEntryNotReady from err
+    except ClientError as err:
         raise ConfigEntryNotReady from err
-    except Exception:  # noqa: BLE001
-        _LOGGER.error("Authentication Error")
-        return False
 
-    hass.data[DATA_CONFIG_ENTRY_LOCK] = asyncio.Lock()
-    hass.data[CONFIG_ENTRY_IS_SETUP] = set()
+    pointSession = PointSession(auth)
 
-    await async_setup_webhook(hass, entry, session)
-    client = MinutPointClient(hass, entry, session)
-    hass.data.setdefault(DOMAIN, {}).update({entry.entry_id: client})
+    client = MinutPointClient(hass, entry, pointSession)
     hass.async_create_task(client.update())
+    hass.data[DOMAIN][entry.entry_id] = client
+
+    hass.data[DOMAIN][DATA_CONFIG_ENTRY_LOCK] = asyncio.Lock()
+    hass.data[DOMAIN][CONFIG_ENTRY_IS_SETUP] = set()
+
+    await async_setup_webhook(hass, entry, pointSession)
+    # Entries are added in the client.update() function.
+    # await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
     return True
 
 
-async def async_setup_webhook(hass: HomeAssistant, entry: ConfigEntry, session):
+async def async_setup_webhook(
+    hass: HomeAssistant, entry: PointConfigEntry, session: PointSession
+) -> None:
     """Set up a webhook to handle binary sensor events."""
     if CONF_WEBHOOK_ID not in entry.data:
         webhook_id = webhook.async_generate_id()
         webhook_url = webhook.async_generate_url(hass, webhook_id)
-        _LOGGER.info("Registering new webhook at: %s", webhook_url)
+        _LOGGER.warning("Registering new webhook at: %s", webhook_url)
```

Comment:
Why should this be logged at the warning level?

Reply:
Well I thought it might be spammy to see this info on info-level. Perhaps debug is better..
```suggestion
        _LOGGER.debug("Registering new webhook at: %s", webhook_url)
```